rest-api/src/test/java/com/cloudbees/workflow/rest/endpoints/JobAndRunAPITest.java

```diff
@@ -281,7 +332,28 @@ private void assertBasicRunInfoOkay(WorkflowJob job, JenkinsRule.WebClient webCl
         Assert.assertEquals(1, workflowRuns.length);
         RunExt runExt = workflowRuns[0];
 
-        assertRunInfoOkay(job, runExt);
+        assertRunPassesSanity(job.getLastBuild(), runExt, false);
+        assertRunInfoOkay(job, runExt, 5, 7, 9);
+    }
+
+    private void assertRunPassesSanity(WorkflowRun myRun, RunExt runExt, boolean testStageChildren) {
+        long stageRunTime = 0;
+        long stagePauseTime = 0;
+        if (runExt.getStages() != null && runExt.getStages().size() > 0) {
+            for (StageNodeExt stage : runExt.getStages()) {
+                assertStageInfoOkay(stage, testStageChildren);
+                stageRunTime += stage.getDurationMillis();
+                stagePauseTime += stage.getPauseDurationMillis();
+            }
+        }
+        Assert.assertTrue("Run has zero duration!", runExt.getDurationMillis() > 0);
+        Assert.assertTrue(String.format("Run duration (%d) ms is shorter than sum of stage runtimes (%d ms), this is insane!",
+                    runExt.getDurationMillis(), stageRunTime),
+                runExt.getDurationMillis() >= stageRunTime);
```

Comment:
Absolutely not - then an assertion failure will break the main method, and the few cases where it would happen indicate something anomalous like a clock time changing.